src/MeshTransfer.f90

```diff
@@ -14,34 +14,848 @@
+  ABSTRACT INTERFACE
+    SUBROUTINE mt_trnsfr_absintfc(this,data_in,data_out)
+      IMPORT :: MeshTransfer_base,SRK
+      CLASS(MeshTransfer_base),INTENT(INOUT) :: this
+      REAL(SRK),INTENT(IN) :: data_in(:)
+      REAL(SRK),INTENT(INOUT),ALLOCATABLE :: data_out(:)
+    ENDSUBROUTINE mt_trnsfr_absintfc
+  ENDINTERFACE
+
+  TYPE,ABSTRACT,EXTENDS(MeshTransfer_base) :: MeshTransfer_1Dbase
+    !> Enumeration for incoming map type
+    INTEGER(SIK) :: MapType_in=0
+    !> Enumeration for outgoing map type
+    INTEGER(SIK) :: MapType_out=0
+    !> The number of incoming mesh elements (or moments for continuous)
+    INTEGER(SIK) :: nmesh_in=0
+    !> The number of outcoming mesh elements (or moments for continuous)
+    INTEGER(SIK) :: nmesh_out=0
+    !>
+    INTEGER(SIK) :: poly_transfer=-1
+    !>
+    REAL(SRK),ALLOCATABLE :: mesh_in(:)
+    !>
+    REAL(SRK),ALLOCATABLE :: mesh_out(:)
+    !>
+    REAL(SRK),ALLOCATABLE :: transferMatrix(:,:)
+    !>
+    TYPE(LinearSolverType_Direct) :: transferLS
+!
+!List of type-bound procedures
+    CONTAINS
+      !> @copybrief MeshTransfer::init_1Dbase
+      !> @copydetails MeshTransfer::init_1Dbase
+      PROCEDURE,PASS :: init_base => init_1Dbase
+      !> @copybrief MeshTransfer::clear_1Dbase
+      !> @copydetails MeshTransfer::clear_1Dbase
+      PROCEDURE,PASS :: clear => clear_1Dbase
+      !> @copybrief MeshTransfer::clear_1Dbase
+      !> @copydetails MeshTransfer::clear_1Dbase
+      PROCEDURE,PASS :: transfer => transfer_1Dbase
+  ENDTYPE MeshTransfer_1Dbase
+
+  TYPE,EXTENDS(MeshTransfer_1Dbase) :: MeshTransfer_1DCart
+!
+!List of type-bound procedures
+    CONTAINS
+      !> @copybrief MeshTransfer::init_1DCart
+      !> @copydetails MeshTransfer::init_1DCart
+      PROCEDURE,PASS :: init => init_1DCart
+  ENDTYPE MeshTransfer_1DCart
+
+  TYPE,EXTENDS(MeshTransfer_1Dbase) :: MeshTransfer_1DCyl
+!
+!List of type-bound procedures
+    CONTAINS
+      !> @copybrief MeshTransfer::init_1DCyl
+      !> @copydetails MeshTransfer::init_1DCyl
+      PROCEDURE,PASS :: init => init_1DCyl
+  ENDTYPE MeshTransfer_1DCyl
 !
 !===============================================================================
   CONTAINS
 !
 !-------------------------------------------------------------------------------
+!> @brief
+!> @param this  The mesh transfer object
+!>
+!> This routine
+!>
+    SUBROUTINE init_1Dbase(this,pList)
+      CLASS(MeshTransfer_1Dbase),INTENT(INOUT) :: this
+      CLASS(ParamType),INTENT(IN) :: pList
+
+      INTEGER(SIK) :: i
+      TYPE(StringType) :: tmpstr
+
+      REQUIRE(pList%has('map_in'))
+      REQUIRE(pList%has('map_out'))
+
+      CALL pList%get('map_in',tmpstr)
+      SELECTCASE(CHAR(tmpstr))
+        CASE('CONTINUOUS')
+          this%MapType_in=MapType_CONTINUOUS
+        CASE('POINT')
+          this%MapType_in=MapType_POINT
+        CASE('VOLUME')
+          this%MapType_in=MapType_VOLUME
+        CASE DEFAULT
+          WRITE(*,*) "invalid incomming map type"
```

Comment:
Change to DBC, no ```WRITE(*,*)```

Reply:
I added the exception handler.  For initiialization, I think we should give clear output errors.  Everything after init, I think DBC is good enough.  That's just my opinion but I think its a good balance of user friendliness and robust testing.  At least this is the standard I was going to propose.  Thoughts?

Reply:
For me the question has always boiled down to:

Do I expect this branch that is an error to happen during normal/production execution, perhaps as a corner case? Or as something the user can get to by modifying the input?

Or is this error really for a developer (e.g. a lot of the ```IF(.NOT.isInit)``` or ```unknown modular ray tracing enumeration```) indicating their usage of the provider is incorrect/not supported.

exception handler is fine.

Reply:
So I'm thinking that I could imagine the input, specifically through decorators, the transfer type could be defined by a user.  You're probably right that these errors were a stretch but I'd rather error on the side of caution at give a better error message than just crash the code immediately.